gradle-plugin/src/proguard/gradle/ProGuardTask.java

```diff
@@ -63,36 +67,45 @@
     private static final String CONFIGURATION_FILE_NAME_PREFIX      = "/proguard/gradle/proguard-";
     private              String resolvedConfigurationFileNamePrefix = getProject().file(CONFIGURATION_FILE_NAME_PREFIX).toString();
 
+    @Inject
+    protected abstract ObjectFactory getObjectFactory();
+
+    @Nested
+    public ProGuardTaskConfiguration getTaskConfiguration() {
+        return ProGuardTaskConfiguration.create(getConfigurationProvider(), getObjectFactory());
```

Comment:
I don't understand why you create a new configuration object for each call here. Why not create the configuration object once.

Reply:
Fixed